src/main/java/com/spingboot/demo/repository/ProductRepository.java

```diff
@@ -1,9 +1,20 @@
 package com.spingboot.demo.repository;
 
 import com.spingboot.demo.domain.Product;
+import com.spingboot.demo.domain.dto.ProductResponseDto;
+import java.util.List;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface ProductRepository extends JpaRepository<Product, Long> {
+
+    @Query("SELECT new com.spingboot.demo.domain.dto.ProductResponseDto(p.productId) "
+            + "FROM Review r "
+            + "JOIN r.product p "
+            + "GROUP BY p.id "
+            + "ORDER BY COUNT(p) DESC, p.id DESC")
+    List<ProductResponseDto> findMostCommented(Pageable pageable);
```

Comment:
Optional: try to use JPA method naming to replace the query here and in each repository. Spring will give you hints as you would start typing the method name